src/esphomelib/time/rtc_component.h

```diff
@@ -0,0 +1,67 @@
+//
+//  rtc_component.h
+//  esphomelib
+//
+//  Created by Brad Davidson on 06.06.18.
+//  Copyright © 2018 Otto Winter. All rights reserved.
+//
+
+#ifndef ESPHOMELIB_TIME_RTC_COMPONENT_H
+#define ESPHOMELIB_TIME_RTC_COMPONENT_H
+
+#include "esphomelib/component.h"
+#include "esphomelib/defines.h"
+#include <stdlib.h>
+#include <time.h>
+
+#ifdef USE_RTC_COMPONENT
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace time {
+
+// A more user-friendly version of struct tm from time.h
+struct EsphomelibTime {
+  int second;
+  int minute;
+  int hour;
+  int day_of_week;
+  int day_of_month;
+  int day_of_year;
+  int month;
+  int year;
+  int is_dst;
+  time_t time;
+};
```

Comment:
For the methods that `lambda` users would use, I think we should definitely have docs (I know this is a WIP PR, but just a heads up😉). For *example*:

```cpp
struct EsphomelibTime {
  /// The second of the minute from 0 to 59.
  int second;
  /// The minute of the hour from 0 to 59.
  int minute;
  /// The hour of the day from 0 to 23.
  int hour;
  /// The day of the week from 1 (sunday) to 7 (saturday).
  int day_of_week;
  /// The day of the month from 1 to 31.
  int day_of_month;
  /// The day of the year from 1 to 366.
  int day_of_year;
  /// The month of the year from 1 (January) to 12 (December)
  int month;
  /// The year, for example 2018
  int year;
  /// Whether daylights saving's time is active for this time zone at the moment.
  int is_dst;
  /// The underlying C time structure <--- Is this required? It doesn't seem to be used anywhere and essentially just doubles the size of the struct.
  time_t time;
};
```

Reply:
Will do. I'm not super familiar with doxygen so I appreciate the pointers on where/how to comment such that it will show up in the docs.

Reply:
Oh, and that's `time_t` (epoch time) not `struct tm`. It's nice to have in there so you can do basic time math (total seconds between events, etc).